### ISSUE TITLE
AWS::StepFunctions::StateMachine - (JSONata) false-positive finding E3601 for Arguments in task

### ISSUE THREAD

**Issue by ianmetcalf** [aws-cloudformation/cfn-lint#4398]
### CloudFormation Lint Version

1.47.0

### What operating system are you using?

Arch Linux

### Describe the bug

JSONata allows using a string for Arguments on a task state as long as it evaluates to a JSON text:

> The value of "Arguments" MUST be a JSON text, or a JSONata string that evaluates to a JSON text.

https://states-language.net/#arguments-and-output

Currently cfn-lint errors on this with `E3601 '...' is not of type 'object'`

### Expected behavior

Not to error

### Reproduction template

```yaml
AWSTemplateFormatVersion: 2010-09-09

Resources:
  StateMachineRole:
    Type: 'AWS::IAM::Role'
    Properties:
      AssumeRolePolicyDocument:
        Statement:
          - Effect: Allow
            Principal:
              Service:
                - states.amazonaws.com
            Action:
              - 'sts:AssumeRole'

  StateMachine:
    Type: 'AWS::StepFunctions::StateMachine'
    Properties:
      RoleArn: !GetAtt StateMachineRole.Arn
      Definition:
        QueryLanguage: JSONata
        StartAt: SetVariables
        States:
          SetVariables:
            Task: Pass
            Assign:
              isTruncated: false
              continuationToken: ''
            Next: ListObjects

          ListObjects:
            Type: Task
            Resource: 'arn:aws:states:::aws-sdk:s3:listObjectsV2'
            Arguments: >-
              {%
                $merge([
                  {
                    "Bucket": "<s3 bucket>"
                  },
                  $isTruncated ? {
                    "ContinuationToken": $continuationToken
                  } : {}
                ])
              %}
            Assign:
              isTruncated: '{% $states.result.IsTruncated %}'
              continuationToken: '{% $states.result.IsTruncated ? $states.result.NextContinuationToken : "" %}'
            Next: CheckTruncated

          CheckTruncated:
            Type: Choice
            Choices:
              - Condition: '{% $isTruncated %}'
                Next: ListObjects
            Default: Complete

          Complete:
            Type: Pass
            End: true
```
